webapp/src/redux/hooks/useStudySynthesis.ts

```diff
@@ -37,21 +37,24 @@ export default function useStudySynthesis<T>(props: UseStudySynthesisProps<T>):
   const dispatch = useAppDispatch();
   const [status, setStatus] = useState<TPromiseStatus>(PromiseStatus.Idle);
   const [error, setError] = useState<Response["error"]>();
+  const [hasAttempted, setHasAttempted] = useState(false);
```

Comment:
dont know for the other use cases if this could be problematic @skamril If you think so I can add a "retry" logic, with a `MAX_RETRY` limit